Doesn't build with GHC-8.6.3

When trying to build the project I get the following error:

```
src/Control/Teardown/Internal/Core.hs:171:10: error:
    • Illegal nested constraint ‘(TypeError ...)’
      (Use UndecidableInstances to permit this)
    • In the instance declaration for ‘IResource [(Text, IO ())]’
    |
171 | instance Ty.TypeError ('Ty.Text "DEPRECATED: Execute a 'newTeardown' call per allocated resource")
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

--- a/src/Control/Teardown/Internal/Core.hs
+++ b/src/Control/Teardown/Internal/Core.hs
@@ -3,6 +3,7 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE UndecidableInstances #-}
 module Control.Teardown.Internal.Core
   ( emptyTeardown
 

